### PR TITLE
fix statically linked build issue for aarch

### PR DESCRIPTION
**Pull request by rajatjindal**
The aarch cross image was using old version of Ubuntu. I had to switch to using image marked as 'edge' in cross-rs repo to fix this.

a new version of cross has not been released in a long time, but using a sha256 of image should be safe until we have a better alternative.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 env:
-  RUST_VERSION: 1.71
+  RUST_VERSION: 1.73
```
Comment (vdice) [inline .github/workflows/release.yml:14]: What is the reason for the bump here?  It's drifting from the [current MSRV](https://github.com/fermyon/spin/blob/main/Cargo.toml#L15).

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -73,6 +73,58 @@ jobs:
           name: spin-ubuntu-latest
           path: target/release/spin
 
+  build-rust-static:
```
Comment (vdice) [inline .github/workflows/build.yml:76]: Any reason we're not using a `matrix` strategy for these two, to slim down the code footprint?

Comment (rajatjindal) [reply .github/workflows/build.yml:76]: no specific reason. i can change it to use matrix strategy.

Comment (rajatjindal) [reply .github/workflows/build.yml]: this is now done

**Review by vdice — APPROVED** [review]

**Pull request merged**